static/js/components/VideoCard_test.js

```diff
@@ -46,16 +53,34 @@ describe('VideoCard', () => {
     it(`video controls ${expect(shouldShow)} be shown for ${testDescriptor}`, () => {
       let isAdmin = adminPermissionSetting;
       let wrapper = renderComponent({isAdmin: isAdmin});
-      assert.equal(wrapper.find(".actions a.edit-link").exists(), shouldShow);
+      //wrapper.find(".actions").children().at(0).children.at(0).children.at(0).children()
+      let menuItems = wrapper.find("Menu").props().menuItems;
+      assert.equal(menuItems.length, (shouldShow ? 3 : 1));
+      if (menuItems.length === 3) {
```

Comment:
as a general rule i think we should avoid forked logic in our tests. we can change up the variables that we're using to compose the test

```javascript
  [
    [false, ['Share'], 'user without admin permissions'],
    [true, ['Share', 'Edit', 'Save on Dropbox'], 'user with admin permissions']
  ].forEach(([adminPermissionSetting, expectedControlLabels, testDescriptor]) => {
    it(`${testDescriptor} should be shown ${expectedControls.length} option(s) for video controls`, () => {
```

then the test case can assert that each item in `expectedControlLabels` matches a label in `menuItems`. might be a little too fancy but that was the first solution i thought of. LMK what you think

Reply:
OK